app/lib/simulator/Simulator.coffee

```diff
@@ -95,7 +95,9 @@ module.exports = class Simulator extends CocoClass
   commenceSingleSimulation: ->
     @listenToOnce @god, 'infinite-loop', @handleSingleSimulationInfiniteLoop
     @listenToOnce @god, 'goals-calculated', @processSingleGameResults
-    @god.createWorld {spells: @generateSpellsObject()}
+    @generateSpellsObject()
+        .then((spell) => @god.createWorld {spells: spell})
```

Comment:
I like this! There is another method `commenceSimulationAndSetupCallback` lower that needs this same change.

Reply:
thanks! but i have to say although `Spells.coffee` is not needed for `simulation` , it needed for battle with other cpp players because it seems we submit both self code and opposite code when we click `run` and the oppsite code is direct run into `Spells.coffee` :joy: